### PR TITLE
Project hierarchy cleanup

### PR DESCRIPTION
**Pull request by bocekm**
- change the directory structure to make it easier and more intuitive to find something.
- rename some files so the filename reflects more the purpose of the file
- add missing simple (aka old) style report xsls
- remove unused files (e.g. common/post_scripts.txt, preuputils/version.py, scripts/rpm-install.sh)

The branch was cloned out to a RHEL 6 machine, installed (python setup.py install), and running the 'preupg' command went smoothly.

### CODE REVIEW

**Comment by bocekm** [comment]
[test]

File: MANIFEST.in [hunk 1]
```diff
@@ -1,27 +1,19 @@
...
-recursive-include man *
-recursive-include preuputils *
-recursive-include postupgrade.d *
-recursive-include scripts *.sh
+include           bin/*
+recursive-include data *
+recursive-include doc *
+recursive-include etc *
+include           man/*
+include           preupg/*.py
```
Comment (phracek) [inline MANIFEST.in]: Moving folder from preup to preupg will have an affect to redhat-upgrade-tool. Be carefull.
I guess, preup can stay as before.

File: preupg/conf.py [hunk 2]
```diff
@@ -1,48 +1,5 @@
...
-# enable debugging
-debug = False
-
-# id of run in frontend
-id = 123456
-
-# prefix of tag in fccdf files
-xccdf_tag = "xccdf_preupg_rule_"
-"""
-
```
Comment (phracek) [inline preupg/conf.py]: This is required for cli.py which uses these values as a default. Otherwise we have to defined the defaults somewhere else.

File: preupg/settings.py [hunk 3]
```diff
@@ -11,37 +11,37 @@
...
+html_result_name = result_prefix + '.html'
 
-# base name of custom xsl stylesheet
 xsl_sheet = "xccdf-report.xsl"
 
-old_xsl_sheet = "preup.xsl"
-
 share_dir = "/usr/share"
 # sources delivered by preupgrade assistant package
 source_dir = os.path.join(share_dir, prefix)
```
Comment (phracek) [inline preupg/settings.py]: prefix is not needed anymore. premigrate will be get rid off.

File: preupg/settings.py [hunk 4]
```diff
@@ -92,27 +89,21 @@
...
 file_list_rules = "list_rules"
 
-# path to file with definitions of common scripts
-post_script = os.path.join(common_dir, "post_scripts.txt")
-
 # kickstart and postupgrade.d directories
-preupgrade_dirs = [dirty_conf_dir, clean_conf_dir,
-                   'hooks', 'kickstart', postupgrade_dir, 'common',
+preupgrade_dirs = [dirty_conf_dir, clean_conf_dir, 'hooks', 'data/templates',
+                   "data/" + postupgrade_dir, 'data/preassesment',
```
Comment (phracek) [inline preupg/settings.py]: This is totaly wrong. !!!! These directories are created by preupg binary before an assessment in /root/preupgrade/ directory. The kickstart directory is used by preupg-kickstart-generated file and contains several files for The Kickstart file. It has an affect to modules. This will cause a several problems. Common is obsolete definitelly.

File: preupg/settings.py [hunk 5]
```diff
@@ -173,13 +164,15 @@
 openssl_command = "openssl x509 -text -in {0} | grep -A1 1.3.6.1.4.1.2312.9.1"
 
 UPGRADE_PATH = ""
-KS_DIR = os.path.join(result_dir, 'kickstart')
-KS_TEMPLATE = 'default.ks'
-KS_TEMPLATE_POSTSCRIPT = 'finish.sh'
-KS_TEMPLATES = [KS_TEMPLATE, KS_TEMPLATE_POSTSCRIPT]
+KS_DIR = os.path.join(assesment_results_dir, 'kickstart')
+KS_FILENAME = 'migration.ks'
```
Comment (phracek) [inline preupg/settings.py]: This is mentioned in several Knowledge Base article. The change will have an affect to change the articles.
I would prefer to use preupgrade.ks filename as it was. Also QA tests will fail.

File: preupg/xmlgen/compose.py [hunk 6]
```diff
@@ -306,15 +305,17 @@ def indent(cls, elem, level=0):
...
+                            settings.xccdf_template)
 
     @classmethod
     def get_xml_tree(cls):
         template_file = ComposeXML.get_template_file()
         try:
             target_tree = ElementTree.parse(template_file).getroot()
         except IOError:
-            print('Problem with reading template.xml file')
+            print('Problem with reading {0} file'
```
Comment (phracek) [inline preupg/xmlgen/compose.py]: {0} will have a problem on python-2.4 which is used on RHEL-5

File: setup.py [hunk 7]
```diff
@@ -5,37 +5,44 @@
...
 project_author_email = "phracek@redhat.com"
 project_description = "Preupgrade assistant"
 package_name = "%s" % project_name
 package_module_name = project_name
 package_version = VERSION
 
-script_files = ['preupg', 'premigrate', 'preup_ui_manage',
-                'preupg-xccdf-compose', 'preupg-create-group-xml',
-                'preupg-content-creator', 'preupg-kickstart-generator']
+script_files = ['bin/preupg', 'bin/premigrate', 'tools/preupg-content-creator',
```
Comment (phracek) [inline setup.py]: Get rid off premigrate.

I would prefer to look on another projects where the binaries are located? In root of the project or in bin directory.

Ask someone.

File: setup.py [hunk 8]
```diff
@@ -5,37 +5,44 @@
...
+        ['preupg/ui/report/fixtures/initial_data.json'],
+    '/usr/share/preupgrade/':
+        ['common.sh', 'doc/README', 'doc/README.kickstart', 'doc/README.ui']
 }
 
-# recursively add templates and static
-paths = ['preup_ui/templates/', 'preup_ui/static/', 'preup_ui/lib/']
+# Include relative path to dirs with non-python files - these will be added
+# to the python module directory with the same relative path
+paths = ['preupg/ui/templates/', 'preupgui/static/']
```
Comment (phracek) [inline setup.py]: Here is a typo.

File: tests/test_xml.py [hunk 9]
```diff
@@ -6,22 +6,14 @@
 import stat
 import tempfile
 
-from xml.etree import ElementTree
-try:
-    from xml.etree.ElementTree import ParseError
-except ImportError:
-    from xml.parsers.expat import ExpatError as ParseError
-
```
Comment (phracek) [inline tests/test_xml.py]: this is needed for XML generation. Removal will have an affect to module generation.

File: MANIFEST.in (hunk 1 above)
Comment (bocekm) [inline MANIFEST.in]: I'll try to find out if it's not too late to change the redhat-upgrade-tool. It's just two lines to change there.

File: preupg/settings.py (hunk 3 above)
Comment (bocekm) [inline preupg/settings.py]: I agree, however I would get rid of the "premigrate" stuff in another branch/pull request. It would be too much changes for this one pull request.

File: preupg/settings.py (hunk 4 above)
Comment (bocekm) [inline preupg/settings.py]: Ok, I'll take a closer look at it and fix it.

File: preupg/xmlgen/compose.py (hunk 6 above)
Comment (bocekm) [inline preupg/xmlgen/compose.py]: I can see it's used a lot in the other code. However, % instead of format is used also a lot.
But to be backwards compatible with python 2.4, I'll switch this to %.

File: setup.py (hunk 7 above)
Comment (bocekm) [inline setup.py]: - I have taken a look into The hacker's guide to Python, where the author recommends the bin directory.
- I'll get rid of all the "premigrate" stuff in one of the subsequent pull requests to not have this one huge.
- I plan to let setup.py create the preupg "binary". It will not be any serious change. I'll do it in on of the subsequent pull requests. Then we won't have the bin directory at all.

File: tests/test_xml.py (hunk 9 above)
Comment (bocekm) [inline tests/test_xml.py]: I don't understand how this change can affect anything. The imports I removed are not used in this test module at all, neither ElementTree, nor ParseError.

File: MANIFEST.in (hunk 1 above)
Comment (phracek) [inline MANIFEST.in]: Thanks

File: preupg/settings.py (hunk 3 above)
Comment (phracek) [inline preupg/settings.py]: ok

File: preupg/settings.py [hunk 10]
```diff
@@ -70,20 +71,16 @@
 # name of dir with common files
 common_name = "common"
 
-# default directory is /var/tmp/preupgrade/common
-# absolute path to dir with common files
-common_dir = os.path.join(share_dir, prefix, common_name)
-
 # path to file with definitions of common scripts
-common_script = os.path.join(common_dir, "scripts.txt")
+common_scripts = os.path.join(data_dir, "preassesment", "scripts.txt")
```
Comment (phracek) [inline preupg/settings.py]: I guess, typo is there assessment is right.

File: preupg/settings.py [hunk 11]
```diff
@@ -92,27 +89,21 @@
...
+preupgrade_dirs = [dirty_conf_dir, clean_conf_dir, 'hooks', 'data/templates',
+                   "data/" + postupgrade_dir, 'data/preassesment',
                    'preupgrade-scripts', 'noauto_postupgrade.d']
 
 
 PREUPG_README = 'README'
-readme_files = {'README': PREUPG_README,
-                'README.kickstart': os.path.join('kickstart', 'README'),
+readme_files = {'doc/README': PREUPG_README,
+                'doc/README.kickstart': os.path.join('kickstart', 'README'),
```
Comment (phracek) [inline preupg/settings.py]: It has to checked if code still works fine.

File: preupg/utils.py [hunk 12]
```diff
@@ -783,10 +784,9 @@ def update_variables(self, result_dir, result_name, xml_result_name, html_result
     @staticmethod
     def get_xsl_stylesheet(old_style=False):
         """Return full XSL stylesheet path"""
-        if old_style:
-            return os.path.join(settings.share_dir, "preupgrade", "xsl", "old_style", settings.old_xsl_sheet)
-        else:
-            return os.path.join(settings.share_dir, "preupgrade", "xsl", settings.xsl_sheet)
+        subfolder = "simple_style" if old_style else "complex_style"
```
Comment (phracek) [inline preupg/utils.py]: This will rise a traceback on RHEL-5.

File: tests/test_xml.py (hunk 9 above)
Comment (phracek) [inline tests/test_xml.py]: ok, I am fine with that. Get rid off it.

**Comment by phracek** [comment]
https://github.com/upgrades-migrations/redhat-upgrade-tool/issues/15

**Comment by bocekm** [comment]
[test]

**Comment by bocekm** [comment]
[test]

**Comment by bocekm** [comment]
[test]

**Comment by bocekm** [comment]
[test]

**Comment by bocekm** [comment]
[test]

**Comment by bocekm** [comment]
[test]

**Comment by bocekm** [comment]
[test]

**Comment by bocekm** [comment]
[test]

**Comment by bocekm** [comment]
[test]

File: MANIFEST.in (hunk 1 above)
Comment (bocekm) [inline MANIFEST.in]: The code will be changed in redhat-upgrade-tool, there is no problem with that. Correct dependency must be specified in the r-u-t spec file.
https://github.com/upgrades-migrations/redhat-upgrade-tool/issues/15

File: preupg/conf.py (hunk 2 above)
Comment (bocekm) [inline preupg/conf.py]: Agreed with @phracek that this may be left removed.

File: preupg/settings.py (hunk 10 above)
Comment (bocekm) [inline preupg/settings.py]: Yes, thanks. It's wrong like this on more places. I'll fix that.

File: preupg/settings.py (hunk 11 above)
Comment (bocekm) [inline preupg/settings.py]: This needs to be reverted. The two readmes are copied to /usr/share/preupgrade by setup.py directly, not to the doc subfolder.

File: preupg/utils.py (hunk 12 above)
Comment (bocekm) [inline preupg/utils.py]: True, ternary operator works since py2.5

File: tools/preupg-ui-manage [hunk 13]
```diff
@@ -1,10 +1,9 @@
 #!/usr/bin/python2
 import os
 import sys
-import preup_ui.libpath
 
 if __name__ == "__main__":
-    os.environ.setdefault("DJANGO_SETTINGS_MODULE", "preup_ui.settings")
+    os.environ.setdefault("DJANGO_SETTINGS_MODULE", "preup.ui.settings")
```
Comment (pirat89) [inline tools/preupg-ui-manage]: Typo: preupg.ui.settings

File: preup_ui/report/admin.py [hunk 14]
```diff
@@ -1,5 +0,0 @@
-# -*- coding: utf-8 -*-
-
-from preup_ui.utils.admin import register_admin_module
-
-register_admin_module('preup_ui.report.models')
```
Comment (phracek) [inline preup_ui/report/admin.py]: Why did you remove it? It is not needed?

**Comment by pirat89** [comment]
Just note, that I havn't review UI part of this change. Could you someone do that?

**Comment by bocekm** [comment]
1. Will add installation of LICENSE by setup.py.
2. I agree, that README, README.ui, README.kickstart shouldn't be copied to /usr/share/preupgrade.
    However I feel that changing it now would be out of scope of this pull request. I tried to keep this one just to changes to the hierarchy/structure of the repository. There are more problems, what is copied where during installation or during execution of the tool. I'll address that in a separate branch/pull request. In the meantime, I'm adding this README idea to [project page](https://github.com/upgrades-migrations/preupgrade-assistant/projects/1).

**Comment by pirat89** [comment]
When you merge this branch, you have to update rhelup - at least imports

**Comment by bocekm** [comment]
@pirat89 to your question about UI. Yes, it works. To make it running without an rpm, use the commands similar to those listed [here](https://gist.github.com/bocekm/ed37931200c6309b9223ca5d68c7fb36).

**Comment by bocekm** [comment]
This pull request is ready to be merged. I'll do that on Nov 21 with the help of @pirat89.
